### PR TITLE
Fix packager LSP behavior

### PR DESCRIPTION
**Pull request by aadi-stripe**
Fixes jump-to-definition behavior in packager. For un-enumerated imports -- i.e. where the top-level namespace of the package is directly aliased into the importing package's namespace -- the current behavior for jump-to-def is to jump to the package file of the import itself. However, it jumps to the last export statement of the package file, which is confusing. Instead, we make it jump to the beginning of the file.

### Motivation

Better usability. Note that ideally, we'd want to standardize the behavior of jump-to-def regardless of whether the import is "enumerated" or "un-enumerated", as that's just an implementation detail of the packager. However, that's a little more complex to make work, and given that the current packager design is a stop-gap for the long run anyway, I think this is probably the local maximum we can achieve.

See jump-to-def behavior in [sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Amodule%20PkgOpusFooPriv%0A%20%20module%20Opus%0A%20%20%20%20module%20Foo%0A%20%20%20%20%20%20module%20A%3B%20end%0A%20%20%20%20%20%20module%20B%3B%20end%0A%20%20%20%20end%0A%20%20end%0Aend%0A%0Amodule%20PkgOpusFooPub%0A%20%20module%20Opus%0A%20%20%20%20module%20Foo%0A%20%20%20%20%20%20A%20%3D%20PkgOpusFooPriv%3A%3AOpus%3A%3AFoo%3A%3AA%0A%20%20%20%20end%0A%0A%20%20%20%20module%20Foo%0A%20%20%20%20%20%20B%20%3D%20PkgOpusFooPriv%3A%3AOpus%3A%3AFoo%3A%3AB%0A%20%20%20%20end%0A%20%20end%0Aend%0A%0Amodule%20PkgOpusBar%0A%20%20module%20Opus%0A%20%20%20%20Foo%20%3D%20PkgOpusFooPub%3A%3AOpus%3A%3AFoo%0A%20%20end%0A%0A%20%20Opus%3A%3AFoo%3A%3AA%0A%23%20%20%20%20%20%20%20%5E%5E%5E%20Foo%20points%20to%20line%2017%0A%23%20%20%20%20%20%20%20%20%20%20%20%20%5E%20A%20points%20to%20line%205%0A%20%20Opus%3A%3AFoo%3A%3AB%0A%23%20%20%20%20%20%20%20%5E%5E%5E%20Foo%20points%20to%20line%2017%0A%23%20%20%20%20%20%20%20%20%20%20%20%20%5E%20B%20points%20to%20line%206%0Aend%0A%0A%23%20We%20effectively%20%22fake%22%20the%20location%20of%20line%2017%20to%20point%20to%20the%20top%20of%20the%20package%20file.) for a simulation.

### Test plan

Tested on Stripe codebase, updated existing LSP test.

ptal: @ngroman

### CODE REVIEW

File: packager/packager.cc [hunk 1]
```diff
@@ -1227,16 +1228,22 @@ class ImportTreeBuilder final {
...
-                const auto &importLoc = useImportLoc ? source.importLoc : core::LocOffsets::none();
+                const auto &moduleLoc = [](auto &source, auto packageLoc) {
+                    // normal or test import
+                    if (source.isTestImport() || source.isNormalImport()) {
+                        return source.isEnumeratedImport ? source.importLoc : core::LocOffsets::none();
+                    }
+
+                    // friend import
+                    return packageLoc->offsets();
+                }(source, packageLoc);
```
Comment (froydnj) [inline packager/packager.cc:1243]: It's pleasing that the local variable here can be declared `const`, but the immediately-evaluated lambda strikes me as non-idiomatic C++.

**Review by froydnj — COMMENTED** [review]
I'll reassign this review to @ngroman , since I think that's who you wanted to review this anyway.

Is it possible to write a test for this?

File: packager/packager.cc (hunk 1 above)
Comment (aadi-stripe) [reply packager/packager.cc:1243]: Can move it to a helper function.

**Comment by aadi-stripe** [comment]
> I'll reassign this review to @ngroman , since I think that's who you wanted to review this anyway.
> 
> Is it possible to write a test for this?

Hmm, I think I might need to update the existing test -- strangely, I thought it was passing for me locally. Will fix.

File: packager/packager.cc (hunk 1 above)
Comment (ngroman) [reply packager/packager.cc:1243]: ++

**Review by ngroman — CHANGES_REQUESTED** [review]
Mostly looks good just needs test fix

File: test/testdata/packager/lsp_features/__package.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 class Simpsons < PackageSpec
     # ^^^^^^^^ symbol-search: "Simpsons", name = "Simpsons", container = "(nothing)"
+    # ^^^^^^^^ symbol-search: "Simpsons", name = "Simpsons", container = "Simpsons"
+    # ^^^^^^^^ symbol-search: "Simpsons", name = "Simpsons", container = "<PackageTests>::Simpsons_Package_Private"
```
Comment (aadi-stripe) [inline test/testdata/packager/lsp_features/__package.rb:8]: Symbol search shows up here rather in an arbitrary export statement.

**Comment by aadi-stripe** [comment]
ptal @ngroman

File: packager/packager.cc (hunk 1 above)
Comment (aadi-stripe) [reply packager/packager.cc]: Fixed

**Review by ngroman — APPROVED** [review]

**Pull request merged**